### PR TITLE
KEYCLOAK-13197 Add OpenJ9 variant of OpenShift guide

### PR DESCRIPTION
**Pull request by hmlnarik**

### CODE REVIEW

**Review by andymunro — DISMISSED** [review]

File: openshift-openj9/master-docinfo.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<productname>{project_name_full}</productname>
+<productnumber>{project_versionDoc}</productnumber>
+<subtitle>For Use with {project_name_full} {project_versionDoc}</subtitle>
```
Comment (iankko) [inline openshift-openj9/master-docinfo.xml]: `s/For Use/For use/g`

File: openshift-openj9/master-docinfo.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+<title>{openshift_openj9_name}</title>
+<release>{project_versionDoc}</release>
+<abstract>
+	    <para>This guide consists of basic information and instructions to get started with {project_name_full} {project_versionDoc} for OpenShift on OpenJ9</para>
+    </abstract>
+<authorgroup>
+    <orgname>Red Hat Customer Content Services</orgname>
+</authorgroup>
+<legalnotice lang="en-US" version="5.0" xmlns="http://docbook.org/ns/docbook">
+  <para> 		Copyright <trademark class="copyright"></trademark> 2019 Red Hat, Inc. 	</para>
```
Comment (iankko) [inline openshift-openj9/master-docinfo.xml]: `s/2019/2020/g`

File: openshift/topics/introduction.adoc [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 === What Is {project_name}?
 {project_name} is an integrated sign-on solution available as a Red Hat JBoss Middleware for OpenShift containerized image. The {project_openshift_product_name} image provides an authentication server for users to centrally log in, log out, register, and manage user accounts for web applications, mobile applications, and RESTful web services.
+{project_name} is only available on the following platforms: *{openshift_image_platforms}*.
```
Comment (iankko) [inline openshift/topics/introduction.adoc]: Suggestion, maybe add cross-reference across the two docs? 

E.g. something aligned with the following: `{project_name} is only available on the following platforms: *{openshift_image_platforms}*. See link_to:{project_name_counterpart_platform_docs_guide_url} documentation for further information about the {project_name_counterpart_platform_image_name}.`

And for both of the guides define `{project_name_counterpart_platform}` as the complement of `["x86_64", "s390x"] - {openshift_image_platforms}`. Similarly `{project_name_counterpart_platform_image_name}`.

WDYT?

**Comment by iankko** [comment]
While at this, there's hardcoded `*_redhat-sso-7/sso72-openshift_*` value of `*_JBOSS_IMAGE_NAME_*` variable in the `topics/reference.adoc` path too, rendering as:

info_env_vars

Would be good to have that one parametrized too if possible.

**Review by iankko — DISMISSED** [review]
Couple of small nitpicks / suggestions, but LGTM in general (as a whole)

@hmlnarik Thanks, Hynek!

**Comment by hmlnarik** [comment]
> While at this, there's hardcoded `*_redhat-sso-7/sso72-openshift_*` value of `*_JBOSS_IMAGE_NAME_*` variable in the `topics/reference.adoc` path too, rendering as:
> 
> info_env_vars
> 
> Would be good to have that one parametrized too if possible.

Could you please introduce a separate JIRA for that?

**Comment by iankko** [comment]
> 
> Could you please introduce a separate JIRA for that?

Right, tracked under [KEYCLOAK-13207](https://issues.redhat.com/browse/KEYCLOAK-13207) now

File: openshift/topics/introduction.adoc (hunk 3 above)
Comment (hmlnarik) [reply openshift/topics/introduction.adoc]: Good idea. Addressed in the updated version.

**Review by iankko — DISMISSED** [review]
The latest version LGTM, thanks Hynek!

**Review by iankko — DISMISSED** [review]
LGTM

File: openshift/topics/before_you_begin.adoc [hunk 4]
```diff
@@ -12,9 +12,9 @@ See the xPaaS part of the https://access.redhat.com/articles/2176281[OpenShift a
 
 [IMPORTANT]
 ====
-The {project_openshift_product_name} image version number 7.0 and 7.1 are deprecated and they will no longer receive updates of image and application templates.
+The {project_openshift_product_name} image version number 7.0 till 7.3 are deprecated and they will no longer receive updates of image and application templates.
```
Comment (stianst) [inline openshift/topics/before_you_begin.adoc]: Till is not commonly used, and until is used in terms of time. I'd go with "between 7.0 and 7.3"

**Review by stianst — CHANGES_REQUESTED** [review]

**Review by stianst — APPROVED** [review]

**Pull request merged**